3 bit descr

Juan-Mateos opened the pull request:
Closes #3 

This PR includes:

* All the BIT raw data (`docx` files with computer science and geography descriptions by school)
* A script to parse these data into two dicts looking up school ids against compsci and geo descriptions  (NB I have also included the outputs in the `inputs/data` folder)
* Getters to read these lookups as well as the BIT master file looking up school ids with names and some other information
* Updated requirements (remember to `pip install -r requirements.txt --upgrade`) before trying to reproduce the code above.
---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [ ] I have requested a code review

Commit:
added python-docx to requirements

Commit:
add raw docx files

Commit:
schools master table

Commit:
parsed descriptions

Commit:
added description getter

Commit:
added and renamed getters

Commit:
script to parse bit data

Commit:
change location of script to parse bit data

Commit:
change script

Commit:
add readme

Jack-0-0 commented on comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py:
```diff
@@ -0,0 +1,43 @@
+BIT_DIRS = ["bit_compsci_descriptions", "bit_geo_descriptions"]
+
+
+def getText(filename: str) -> str:
```
Function `getText` should be in camel case: `get_text` as per the style guidelines

Jack-0-0 commented on comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py:
```diff
@@ -0,0 +1,43 @@
+    """Reads and parses a doc file from BIT"""
+
+    doc = docx.Document(filename)
+    fullText = []
```
Variable `fullText` should be in camel case: `full_text` as per the style guidelines

Jack-0-0 commented on comp_sci_gender_bias/pipeline/text_data/parse_bit_data.py:
```diff
@@ -0,0 +1,43 @@
+
+    doc = docx.Document(filename)
+    fullText = []
+    for para in doc.paragraphs:
+        fullText.append(para.text)
```
Creating `fullText` could be simplified to `fullText = [para.text for para in doc.paragraphs]`

Jack-0-0 commented on comp_sci_gender_bias/getters/school_data.py:
```diff
@@ -0,0 +1,32 @@
+def school_table() -> pd.DataFrame:
+    """Reads a table with school metadata including lookup between ids and names"""
+
+    return pipe(
+        pd.read_csv(f"{PROJECT_DIR}/inputs/data/school_master_table.csv"),
+        lambda df: df.rename(
+            columns={col: re.sub(" ", "_", col.strip().lower()) for col in df.columns}
+        ),
+        lambda df: df.rename(columns={"number": "id"}),
```
Nice use of `pipe` to rename the columns 🙂

Jack-0-0 reviewed (commented):
Thanks for this PR @Juan-Mateos. All the code runs as expected 🙂

I have made a few comments on specific lines of code.

All the docx inputs are being stored on github. Why are we not storing these on S3?

Commit:
addressed PR comments

Juan-Mateos commented:
I addressed the comments above i.e. 
 - [ ] Changed `GetText` name
 - [ ] Removed `for` loop with list comprehension
 - [ ] Re-ran the script and confirmed it works

Jack-0-0 reviewed (approved):
Thanks @Juan-Mateos. This looks good to merge
